### PR TITLE
Cap pluggy 1.3 release

### PR DESCRIPTION
**Pull request by DimedS**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
This PR limits `pluggy` to version below `1.3` due to failing tests in `kedro`.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
Thanks @DimedS!

File: pyproject.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ dependencies = [
     "omegaconf~=2.3",
     "parse~=1.19.0",
     "pip-tools>=6.5,<8",
-    "pluggy~=1.0",
+    "pluggy>=1.0, <1.3",
```
[the remark this answers is not in the document]
Comment (DimedS) [reply pyproject.toml]: Good point, thanks @astrojuanlu. I've made the addition

**Review by ankatiyar — APPROVED** [review]
Thank you!

**Review by SajidAlamQB — APPROVED** [review]
Thank you @DimedS!

**Pull request merged**